test/parallel/test-stream-duplex-destroy.js

```diff
@@ -284,3 +284,17 @@ const assert = require('assert');
   duplex.on('close', common.mustCall());
   duplex[Symbol.asyncDispose]().then(common.mustCall());
 }
+
+(async () => {
+  // Check Symbol.asyncDispose implicitly
+  await using duplex = new Duplex({
+    write(chunk, enc, cb) { cb(); },
+    read() {},
+  });
+  let count = 0;
+  duplex.on('error', common.mustCall((e) => {
+    assert.strictEqual(count++, 0); // Ensure not called twice
+    assert.strictEqual(e.name, 'AbortError');
+  }));
+  duplex.on('close', common.mustCall());
+})().then(common.mustCall(), common.mustNotCall());
```

Comment:
nit: for the sake of simplicity – and to be consistent with most of the codebase – let's not pass `common.mustNotCall()`
```suggestion
})().then(common.mustCall());
```